ledger/participant-integration-api/src/main/scala/platform/store/serialization/Compression.scala

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) 2020 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.platform.store.serialization
+
+import java.io.{ByteArrayOutputStream, InputStream, OutputStream}
+import java.util.zip.{GZIPInputStream, GZIPOutputStream}
+
+import com.codahale.metrics.Histogram
+
+private[platform] object Compression {
+
+  sealed abstract class Algorithm(val id: Option[Int]) {
+    final def compress(uncompressed: Array[Byte], ratio: Histogram): Array[Byte] = {
+      val output = new ByteArrayOutputStream(uncompressed.length)
+      val gzip = compress(output)
+      try {
+        gzip.write(uncompressed)
+      } finally {
+        gzip.close()
+      }
+      val compressed = output.toByteArray
+      output.close()
+      ratio.update(compressed.length * 100 / uncompressed.length)
```

Comment:
I'd turn the ratio around

Reply:
Ultimately I'll probably do what you originally did and simply record compressed and uncompressed size separately, but thanks.

Reply:
Addressed by https://github.com/digital-asset/daml/pull/8039/commits/1165b8bf5e604f18427f3998eb566ef098696966